### ISSUE TITLE
 CL_INVALID_BUFFER_SIZE

### ISSUE THREAD

**Issue by carboneer** [fireice-uk/xmr-stak-amd#147]
Hi , i'm trying to use a HD 6990 4Gb Radeon dual GPU but when i try to lunch the xmr-stark-amd.exe i have this error 
 Error CL_INVALID_BUFFER_SIZE when calling clCreateBuffer to create hash scratchpads buffer.

I try to watch the other issues to solve this problem but the new issue doesn't fix it . 
And my main ram is 8gb > GPU ram 4gb .
Can you help me plz ?
[config.txt](https://github.com/fireice-uk/xmr-stak-amd/files/1385422/config.txt)
[clinfo.txt](https://github.com/fireice-uk/xmr-stak-amd/files/1385423/clinfo.txt)

**Comment by carboneer** [comment]
[clinfo.txt](https://github.com/fireice-uk/xmr-stak-amd/files/1385617/clinfo.txt)
And this is my new clinnfo after the new drivers but still the same problem :(

**Comment by elisaado** [comment]
My solution was to change the intensity of both threads to 64, if that works, try 128, if that works try 256 etc. etc. until it doesn't work anymore.

**Comment by carboneer** [comment]
If i change the intensity to 256 and then it's work but in very slow mode less than 300 h/s for my 4 Gpu .
Currently i'm using claymore miner who give me 1,2k h/s ,i hope to find something to fix this error .

**Comment by psychocrypt** [comment]
We are working on that. The next version will fix most of this issues. We
hope to release within the next two weeks.

**Comment by TheIz80** [comment]
Is the Fix there yet?

**Comment by Henhouser** [comment]
Carboneer's problem is the same one as mine, I too have a Radeon HD 6990 and get the CL_INVALID_BUFFER_SIZE error when launching xmr-stak-amd.  I see Psychocrypt mentioned their newer version, is this that xmr-stak all-in-one for NVIDIA AMD and CPU that the xmr-stak-amd page links to?  I had the impression that new one didn't support the AMD 6000 series, none of the other new miners seem to.  Anyone with more info, please leave some of it here, it could change my life to get this working soon.
Edit:  I tried the new one, xmr-stak-win64 and it delivers the same CL_INVALID_BUFFER_SIZE error, plus AMD device not found and backend AMD disabled.  The Pro 6990 driver software I'm using is whql-win10-catalyst-15.7.1-oct30, which is the closest to 15.12 that I can get to install to this card, since that driver version everyone recommends is incompatible.  Claymore can see the 6990 with driver 15.7.1, but Claymore fails to other errors, No ASM binary found for GPU and the like.

**Comment by psychocrypt** [comment]
This is an driver issue. The driver is reporting to less memory to the
miner. You can use the latest release of xmr-stak. The auto generation will
show you the amount of free memory reported to the miner.

**Comment by Henhouser** [comment]
I see, the latest xmr-stak says the amount of memory reported per GPU is 384, I think that ideally it should be 2048.  The driver version that I have is 15.200.1062.0, do you know if there is a driver the 6990 GPU can use that xmr-stak can also use?  Thank you for explaining.

**Comment by Reiz45** [comment]
What @carboneer said fixed my issue with this. Thank you!

**Comment by Reiz45** [comment]
What @elisaado said worked for me. Thanks!
*Edit: This only got my card working. My card is only running 250-350 H/s. If any one figures out the settings to bump it up to 1600 h/s, let me know. 
I'm running an AMD 560 card. Using the Stak AMD program. @Voldearag My intensity setting is at 897ish. Any higher and I get the error message.

**Comment by Henhouser** [comment]
@Reiz45  You changed the intensity settings to 256?  That makes the card hash for me too but at only 316H/s which is less than my CPU gets.  Are you using a Radeon 6990, and what is your hash rate?
Edit: The new xmr-stak-win64 is giving me 455H/s while both GPUs are set to 2048 memory, intensity 256 and worksize 8, that's a bit better than CPU speed.  Pools keep banning me for restarting my miner too much while testing settings.

**Comment by psychocrypt** [comment]
Could everyone please post the output of 'clinfo' and the auto generated
config. I am sill searching why opencl is reporting to less memory. I hope
that I can found the reason with more data.

**Comment by ccernat** [comment]
Very strange with these old cards. I am on WIN10 x64. Tried 13.2, 14.9 and 15.2 drivers for AMD for HD6990x2. As miners tried xmrig, xmr-stak, claymore. Just NOTHING works.

xmr-stak works only with 13.2, but HORRENDOUS hash rates, and only if I keep the intensity at less or equal with 256. These cards should mine at 1,6KH on CryptoNight. I get 150 h/sec LOL. It is unbelievable. Something is really wrong with the drivers and miner software. If I up the intensity, CL_Invalid_Buffer Size.

On driver 13.2 Catalyst xmr-stak reports Pitcairn memory 1613 for each of the GPU's. ( card has a total of 4 GB available). On any other drivers it reports 384 Mb memory the xmr stack autoconfig. It is frustrating as hell. 

If someone has managed to find a working configuration of operating system, drivers, miner brand and configuration for these type of HD cards, please holla at me. Thank you very much!

**Comment by psychocrypt** [comment]
1613 MB sounds good. You need to set the environment variable to use more.
OpenCL allowes by default that only 1/4 of the memory can be used with one
allocation.

**Comment by ccernat** [comment]
Yes, 1613 but with a max intensity of 256 and only 80 h/s per core HD6990 with xmr-stak. If I up the intensity, CL_Invalid_Buffer and miner use only CPU. I am dying to find a solution... Long way till 1,6Kh/sec. I am out of ideas.

**Comment by ccernat** [comment]
It would be nice if you, as a developer, could grab one of these cards for testing. Maybe someone is willing to send you one. Or at least give you access to a computer running of these.

**Comment by ccernat** [comment]
One question: is it possible that not enough power is supplied to the card to cause these problems? Just read that these monster eat around 528W in full throttle mode. My PSU is 650W for the whole system. Hmm...

**Comment by Henhouser** [comment]
Here is my clinfo output and auto generated xmr-stak-win64 settings. https://puu.sh/z8he3/23b1c04f70.zip 
I'm using a Radeon HD 6990 in Windows 7 x64, and an enormous 40GB pagefile just in case that matters.
Thank you for checking this out for us, Psychocrypt.

**Comment by TrueOsiris** [comment]
clinfo https://pastebin.com/j79EMmmm
Any news?

**Comment by psychocrypt** [comment]
please use the lates release https://github.com/fireice-uk/xmr-stak/releases

**Comment by Orsontrius** [comment]
Hello I am using the latest release (v2.2.0) and I am still getting the same problem.
I'm using two Radeon HD 5770's and if I increase the intensity beyond 64, I get the error:
CL_INVALID_BUFFER_SIZE when calling clCreateBuffer to create hash scratchpads buffer.
Using 64 as the intensity it works, but at around 30-40h/s per card

I'm using ubuntu 14.04 and AMD Catalyst 15.9 drivers

**Comment by psychocrypt** [comment]
You are running out of memory. The old opencl driver not allows to allocate
a big chunk of memory.
Remove the file amd.txt and try
https://github.com/fireice-uk/xmr-stak/blob/master/doc/tuning.md#increase-memory-pool

**Comment by Orsontrius** [comment]
I removed amd.txt, set the environment variables as it says to do so, then started the miner.
I still have the same problem
It gives the invalid buffer size error until I lower the intensity to 64

**Comment by Lowerland** [comment]
Autogenerated conf:
[amd.txt](https://github.com/fireice-uk/xmr-stak-amd/files/1844315/amd.txt)

clinfo:
[clinfo.txt](https://github.com/fireice-uk/xmr-stak-amd/files/1844316/clinfo.txt)

tried also setting the environment variables but no luck

@psychocrypt 
It gives the invalid buffer size error no matter what i set for intensity

**Comment by lxnay** [comment]
having the same issue here with the RX570, AMDGPU-PRO 18.10 on Linux.

**Comment by lxnay** [comment]
ethminer says: OpenCL device Ellesmere  has insufficient GPU memory. 76398592  bytes of memory found < 1073739904  bytes of memory required
